core/src/main/java/com/linecorp/armeria/internal/client/dns/DnsUtil.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.client.dns;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+
+import org.slf4j.Logger;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.CommonPools;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.util.TransportType;
+
+import io.netty.buffer.ByteBufUtil;
+import io.netty.channel.EventLoop;
+import io.netty.handler.codec.dns.DnsRecord;
+import io.netty.handler.codec.dns.DnsRecordType;
+import io.netty.resolver.dns.DnsNameResolver;
+import io.netty.resolver.dns.DnsNameResolverBuilder;
+
+public final class DnsUtil {
+
+    private static final List<String> DEFAULT_SEARCH_DOMAINS;
+    private static final int DEFAULT_NDOTS;
+
+    static {
+        try {
+            // TODO(ikhoon): Fork Netty code to avoid reflections for the default options.
+            final EventLoop eventLoop = CommonPools.workerGroup().next();
+            final DnsNameResolver defaultResolver =
+                    new DnsNameResolverBuilder(eventLoop)
+                            .channelType(TransportType.datagramChannelType(eventLoop.parent()))
+                            .build();
+            final Method searchDomainsMethod = DnsNameResolver.class.getDeclaredMethod("searchDomains");
+            searchDomainsMethod.setAccessible(true);
+            final String[] searchDomains = (String[]) searchDomainsMethod.invoke(defaultResolver);
+            DEFAULT_SEARCH_DOMAINS = ImmutableList.copyOf(searchDomains);
+            final Method ndotsMethod = DnsNameResolver.class.getDeclaredMethod("ndots");
+            ndotsMethod.setAccessible(true);
+            DEFAULT_NDOTS = (int) ndotsMethod.invoke(defaultResolver);
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+            throw new Error(e);
```

Comment:
- How about adding a helpful exception message?
- Would it better falling back with an empty search list with 0 ndots with warning logs?